### DISCUSSION TITLE
Recommended approach for taking controller input

### DISCUSSION THREAD

**Discussion by h0uter** [zauberzeug/nicegui#3447] [Ideas / Feature Requests]
![image](https://github.com/user-attachments/assets/f56d738d-3310-4f08-a2e4-f3bd64be1577)

[ui.keyboard](https://nicegui.io/documentation/keyboard) works great to take user input! What is the recommended approach for doing the same using a controller. For example when visiting a NiceGUIwebsite on a SteamDeck?

Buttons that would need to be read :
- left and right joysticks
- bumpers
- A B X Y / triangle square circle cross
- arrow-pad
- etc.

**Comment by rodja** [comment]
As far as I know, gamepads do not fire events in the same way as keyboard and mouse in JavaScript. But there is a [GamePad API](https://www.w3.org/TR/gamepad/) which should work.
